pmd-core/src/main/java/net/sourceforge/pmd/lang/metrics/ParameterizedMetricKey.java

```diff
@@ -65,6 +70,7 @@ public int hashCode() {
      */
     @SuppressWarnings("PMD.SingletonClassReturningNewInstance")
     public static <N extends Node> ParameterizedMetricKey<N> getInstance(MetricKey<N> key, MetricOptions options) {
+        // sharing instances allows using DataMap, which uses reference identity
```

Comment:
I think, `POOL` needs to be a concurrent map, otherwise there might be a chance, that we create two instances for the same... I'll change that before merging